src/formatDate.js

```diff
@@ -8,7 +8,52 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[fromFormat.length - 1]);
+
+  const formattedDateParts = {
+    YYYY: '',
+    MM: '',
+    DD: '',
+  };
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YYYY':
+        formattedDateParts['YYYY'] = dateParts[i].substring(0, 4);
+
+        if (toFormat.includes('YY')) {
+          formattedDateParts['YY'] = dateParts[i].substring(2, 4);
+        }
```

Comment:
The check for 'YY' in 'toFormat' and setting 'formattedDateParts['YY']' should be done outside the switch statement to ensure that 'YY' is set correctly even if 'YYYY' is not part of 'fromFormat'.